doc/source/data/batch_inference.rst

```diff
@@ -1,823 +1,455 @@
+            predictions = self.model(list(batch["data"]), max_length=20, num_return_sequences=1)
+            batch["output"] = [sequences[0]["generated_text"] for sequences in predictions]
+            return batch
 
-        .. button-ref:: /data/examples/batch_inference_object_detection
+    predictions = ds.map_batches(
+        HuggingFacePredictor, 
+        # Require 5 CPUs per actor (so at most 3 can fit per 16 CPU node).
+        num_cpus=5,
+        # 3 actors per node, with 4 nodes in the cluster means ActorPool size of 12.
+        compute=ray.data.ActorPoolStrategy(size=12) 
+        )
+    predictions.show(limit=1)
 
-            Object Detection Batch Inference with PyTorch
+How Ray Data compare to X for offline inference?
+------------------------------------------------
 
-    .. grid-item-card::
-        :img-top: /images/ray_logo.png
-        :class-img-top: pt-2 w-75 d-block mx-auto fixed-height-img
+.. dropdown:: Batch Services: AWS Batch, GCP Batch
 
-        .. button-ref:: /data/examples/ocr_example
+    Cloud providers such as AWS, GCP, and Azure provide batch services to manage compute infrastructure for you. Each service uses the same process: you provide the code, and the service runs your code on each node in a cluster. However, while infrastructure management is necessary, it is often not enough. These services have limitations, such as a lack of software libraries to address optimized parallelization, efficient data transfer, and easy debugging. These solutions are suitable only for experienced users who can write their own optimized batch inference code.
 
-            Batch OCR processing using Ray Data
+    Ray Data abstracts away not only the infrastructure management, but also the sharding your dataset, the parallelization of the inference over these shards, and the transfer of data from storage to CPU to GPU.
 
-    .. grid-item-card::
-        :img-top:  /images/ray_logo.png
-        :class-img-top: pt-2 w-75 d-block mx-auto fixed-height-img
 
-        .. button-ref:: /ray-air/examples/torch_detection
+.. dropdown:: Online inference solutions: Bento ML, Sagemaker Batch Transform
 
-            Fine-tuning an Object Detection Model and using it for Batch Inference
+    Solutions like `Bento ML <https://www.bentoml.com/>`_, `Sagemaker Batch Transform <https://docs.aws.amazon.com/sagemaker/latest/dg/batch-transform.html>`_, or :ref:`Ray Serve <rayserve>` provide APIs to make it easy to write performant inference code and can abstract away infrastructure complexities. But they are designed for online inference rather than offline batch inference, which are two different problems with different sets of requirements. These solutions often don't perform well in the offline case, leading inference service providers like `Bento ML to integrating with Apache Spark <https://modelserving.com/blog/unifying-real-time-and-batch-inference-with-bentoml-and-spark>`_ for offline inference.
```

Comment:
do you want to add a brief explanation why they are suboptimal for offline jobs?